client/src/components/Ticketing/ticketingmanager/Season/SeasonInstancesPage.tsx

```diff
@@ -26,45 +26,60 @@ interface SeasonInstancesProp {
 const SeasonInstancesPage = (props: SeasonInstancesProp) => {
   const navigate = useNavigate();
   const {token} = props;
-  const [seasons, setAllSeasons] = useState<Seasons[]>([]);
+  const [filterSetting, setFilterSetting] = useState('active');
+  const [seasonData, setSeasonData] = useState([]);
+  const [allSeasonData, setAllSeasonData] = useState([]);
+  const [inactiveData, setInactiveData] = useState([]);
+  const [activeData, setActiveData] = useState([]);
 
-  const getAllSeasons = async () => {
-    try {
-      const getAllSeasonsRes = await fetch(
-        process.env.REACT_APP_API_2_URL + '/season',
-        {
-          credentials: 'include',
-          method: 'GET',
-          headers: {
-            'Content-Type': 'application/json',
-            'Authorization': `Bearer ${token}`,
-          },
-        },
-      );
-
-      if (!getAllSeasonsRes.ok) {
-        throw new Error('Failed to fetch all seasons');
-      }
+  useEffect(() => {
+    fetch(process.env.REACT_APP_API_2_URL + '/season/list', {
+      headers: {
+        'Authorization': `Bearer ${token}`,
+      },
+    })
+    .then((response) => response.json())
+    .then((data) => {
+      const inactiveSeasons = [];
+      const activeSeasons = [];
+      const allSeasons = [];
 
-      const seasonsData = await getAllSeasonsRes.json();
-      setAllSeasons(seasonsData);
-    } catch (error) {
-      console.error(error);
-    }
-  };
+      data.forEach((season) => {
+        const allEventsActive = season.events.every((event) => (event.active));
 
-  useEffect(() => {
-    getAllSeasons();
+        if (allEventsActive) {
+          activeSeasons.push(season);
+          allSeasons.push(season);
+        } else {
+          inactiveSeasons.push(season);
+          allSeasons.push(season);
+        }
+      });
+      // Set active/inactive/all data arrays
+      setInactiveData(inactiveSeasons);
+      setActiveData(activeSeasons);
+      setAllSeasonData(allSeasons);
+      // Set Default "Active"
+      setSeasonData(activeSeasons);
+    })
+    .catch((error) => {
+      console.error('Error Fetching Season Data:', error);
+    });
   }, []);
 
-  /**
-   * Based on active/inactive/all
-   *
-   * @param event
-   */
-  const handleEventChange = (event) => {
-    // TODO: handle the season type change when active/inactive is properly implemented
-  };
+  // Group Toggle Display Changes
+  useEffect(() => {
+    if (filterSetting === 'active') {
+      // Active Data Only
+      setSeasonData([...activeData].sort((a, b) => b.seasonId - a.seasonId));
+    } else if (filterSetting === 'inactive') {
+      // Inactive Data Only
+      setSeasonData([...inactiveData].sort((a, b) => b.seasonId - a.seasonId));
+    } else {
+      // All Seasons
+      setSeasonData([...allSeasonData].sort((a, b) => b.seasonId - a.seasonId));
```

Comment:
To reduce some of the operations (the spreading and sorting each time the toggle is pressed), you could move the sorts to lines 59-61.

Reply:
refactored